Client/Api/Builder/ICredentialsProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Zeebe.Client.Api.Builder
+{
+    public interface ICredentialsProvider
+    {
+        // Adds credentials to the request headers.
+        void ApplyCredentials(AuthInterceptorContext context, Metadata metadata);
```

Comment:
I would rather not expose the gRPC related classes, so we are still independent of the implementation.

Reply:
So maybe we can hide them behind an interface? What do you think?